packages/lib/timeFormat.ts

```diff
@@ -6,23 +6,35 @@
  */
 import { localStorage } from "@calcom/lib/webstorage";
 
+const is24hLocalstorageKey = "timeOption.is24hClock";
+
+export const setIs24hClockInLocalStorage = (is24h: boolean) =>
+  localStorage.setItem(is24hLocalstorageKey, is24h.toString());
+
+export const getIs24hClockFromLocalStorage = () => {
+  const is24hFromLocalstorage = localStorage.getItem(is24hLocalstorageKey);
+  if (is24hFromLocalstorage === null) return null;
+
+  return is24hFromLocalstorage === "false" ? false : true;
+};
+
 export const isBrowserLocale24h = () => {
-  const localStorageTimeFormat = localStorage.getItem("timeOption.is24hClock");
+  const localStorageTimeFormat = getIs24hClockFromLocalStorage();
```

Comment:
```suggestion
  const localStorageTimeFormat = !!getIs24hClockFromLocalStorage();
```

Reply:
In this case we don't want to convert null into the boolean as well. If it's null we explicitly want to skip to the logic of trying to infer this from the browsers language.